modules/webgl/src/adapter/webgl-device.ts

```diff
@@ -192,7 +192,8 @@ ${device.info.vendor}, ${device.info.renderer} for canvas: ${device.canvasContex
     }
 
     // Create and instrument context
-    this.canvasContext = new WebGLCanvasContext(this, props);
+    const canvas = props.canvas || props.gl?.canvas;
```

Comment:
We might even change the order here, if there is a gl context it should not be possible to override the canvas. Or even throw an error